lib/json/common.rb

```diff
@@ -637,6 +637,12 @@ def dump(obj, anIO = nil, limit = nil, strict: NOT_SET)
   def self.iconv(to, from, string)
     string.encode(to, from)
   end
+
+  private
+
+  def merge_dump_options(opts, strict: NOT_SET)
+    opts[:strict] = strict if NOT_SET != strict
```

Comment:
This can destructively modify `JSON.dump_default_options`. I addressed that in https://github.com/flori/json/pull/556.